src/Symfony/Component/FeatureToggle/StrategyResult.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\FeatureToggle;
+
+enum StrategyResult
+{
+    case Grant;
+    case Deny;
+    case Abstain;
```

Comment:
I think this naming makes more sense in a feature toggle context. 

```suggestion
    case Enable;
    case Disable;
    case Fallback;
```

Reply:
To me Disable does not make much sense in a Startegy context. It does not Disable a feature. Simply stating "I would not enabling it"

Reply:
Also, "Abstain" means "I can't give a result, ask someone else". Renaming it to "Fallback" may be interpreted as "I can't give a result, I give you my default value".

Reply:
Yeah, could be, to me it reads that fallback is "I can't know for sure, so fallback to the feature default state".

This could be a strategy depending on a request param or env var for instance. If it's not defined fallback to default, if is defined use the value to enable/disable.

In a date period strategy, it doesn't return a fallback, the date is either inside the period or not

Reply:
For the date strategy there is indeed no "abstain" possible. Either in or out. Regarding others, it means "I cannot answer for sure neither grant nor deny" Instead of `Abstain` (which made sense to me because of Security voters) it could be renamed something else. `Neither` for example. But `Fallback` gives too much meaning as to what happens as a consequence. To me, it shouldn't have to care about what will happen only what is the Strategy answer.

And because it cannot answer, I thought "Abstain" makes sense.